Japanese translation of docs/custom_matrix.md

mtei opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
## Description

Add a Japanese translation of docs/custom_matrix.md.

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

mtei commented:
@umi-umi @shelaf @s-show @hsgw よかったらレビューお願いします。

hsgw commented:
#8278 では`matrix`の翻訳は`マトリクス`としています。どちらかに統一したいです。

hsgw commented on docs/ja/custom_matrix.md:
````diff
@@ -0,0 +1,115 @@
+SRC += matrix.c
+```
+
+## マトリクスコードの一部補完
````
```suggestion
## 'lite'
```
以下に説明があるので原文のままで良いと思います。

mtei replied:
目次が、トップに表示されるので、わかりやすくしたいです。
https://docs.qmk.fm/#/ja/newbs_building_firmware

hsgw replied:
確かにそうですね。
コードの`置換`が今ひとつピンとこないので、いっそのこと`実装`でまとめてしまうとわかりやすそうです。

mtei replied:
これは、原文に注文つける案件 😁

mtei commented:
既存の訳は、「マトリックス」なんですねぇ。どうしましょ。

```shell
$ git checkout master
Switched to branch 'master'
$ grep マトリックス *.md | wc -l
      24
$ grep マトリクス *.md | wc -l  
       2

$ git checkout review/qmk/8196_japnese_part_01
Switched to branch 'review/qmk/8196_japnese_part_01'
$ grep マトリックス *.md | wc -l
      30
$ grep マトリクス *.md | wc -l
       2

$ git checkout review/qmk/8278_japanese_hard
Switched to branch 'review/qmk/8278_japanese_hard'
$ grep マトリックス *.md | wc -l            
      24
$ grep マトリクス *.md | wc -l            
      16

$ git checkout -
Switched to branch 'review/qmk/8196_japnese_part_01'
$ grep マトリクス *.md        
_summary.md:   * [LED マトリクス](ja/feature_led_matrix.md)
_summary.md:   * [RGB マトリクス](ja/feature_rgb_matrix.md)
```

mtei commented:
@zvecr,
There is some inconsistency in the notation of 'keyboard directory' in docs/custom_matrix.md.
I translated all of them into 'directory'.

docs/custom_matrix.md L17
```
Add a new file to your keyboard directory:
```

docs/custom_matrix.md L36
```
And implement the following functions in a `matrix.c` file in your keyboard folder:
```

docs/custom_matrix.md L62
```
And implement the following functions in a `matrix.c` file in your keyboard folder:
```

mtei commented:
@zvecr,
I think this table of contents is a bit inconsistent.
Therefore, the Japanese translation translated 'lite' as a partial replacement.
<img width="417" alt="z" src="https://user-images.githubusercontent.com/2170248/76859391-990a5700-689c-11ea-9a1b-38405f707213.png">

mtei commented:
とりあえず、暫定で「マトリックス」で統一して手直ししました。引き続きご意見まってます。

zvecr commented:
On directory vs folder, I don't mind which one we standardise on.

As for the TOC, `partial replacement` seems fine to me. Though we might want to reword the opening sentence as it partially requires the title for it to make sense.

Feel free to align the English docs if it also helps!

shelaf commented on docs/ja/custom_matrix.md:
```diff
@@ -0,0 +1,114 @@
+
+## マトリックスコードの部分的入れ替え
+
+カスタムマトリックスを実装する際、さまざまなスキャン機能のデフォルト実装を提供している定型コードは削減します。
```
```suggestion
カスタムマトリックスを実装する際、定型コードを削減するために、さまざまなスキャン機能のデフォルト実装を提供しています。
```

mtei replied:
読み違えていました。

定型コードの実装の手間を削減するために、デフォルト実装を提供しているってことですね？
提示してもらった訳文は、いまいちそこがはっきりしない気がします。原文が言葉足らずなのかも。

mtei replied:
訳注つけて、すませます。

shelaf commented on docs/ja/custom_matrix.md:
````diff
@@ -0,0 +1,114 @@
+}
+```
+
+## マトリックスコードの全部入れ替え
````
完全な置き換え、みたいな訳かなぁと思います。

mtei replied:
二つのセクションのタイトル、対でそろえて
 * マトリックスコードの部分入れ替え
 * マトリックスコードの全面入れ替え

でしょうか？
Replacement は、「入れ替え」「置き換え」「交換」いろいろ表現がありうるけど、、どれにしたものか。

shelaf commented on docs/ja/custom_matrix.md:
```diff
@@ -0,0 +1,114 @@
+
+## マトリックスコードの部分置き換え
+
+カスタムマトリックスを実装する際、定型コードを削減するために(訳注：定型コードを実装する手間の削減?)、さまざまなスキャン機能のデフォルト実装を提供しています。
```
訳注のところの「？」はなくていいと思います。
直訳だと、
様々なスキャン関数のデフォルト実装を提供し、カスタムマトリックスの実装時に定型コードを削減します。
という感じです。
指摘忘れですがスキャン機能というよりはスキャン関数ではないでしょうか。

mtei replied:
「関数」ですね。
さらに意訳してみました。

mtei commented:
I think this PR is ready to merge.

hsgw, what do you think?

noroadsleft commented:
ありがとうございました！
